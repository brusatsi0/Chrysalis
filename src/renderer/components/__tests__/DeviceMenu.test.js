// -*- mode: js-jsx -*-
/* Chrysalis -- Kaleidoscope Command Center
 * Copyright (C) 2018, 2019  Keyboardio, Inc.
 *
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation, version 3.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from "react";
import { render, fireEvent, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";

import DeviceMenu from "../DeviceMenu";

afterEach(cleanup);
const mockOpenMenu = jest.fn();
const device = {
  urls: { url: "http://www.keyboard.io", name: "Home" },
  displayName: "Keyboardio Model01"
};

test("DeviceMenu renders a button with the proper device name", () => {
  const { getByText } = render(
    <DeviceMenu openBoardMenu={mockOpenMenu} device={device} />
  );

  expect(getByText("Keyboard: Keyboardio Model01")).toMatchSnapshot();
});

test("DeviceMenu button calls openMenu when clicked", () => {
  const { getByText } = render(
    <DeviceMenu openBoardMenu={mockOpenMenu} device={device} />
  );
  fireEvent.click(getByText("Keyboard: Keyboardio Model01"));

  expect(mockOpenMenu).toHaveBeenCalled();
});
