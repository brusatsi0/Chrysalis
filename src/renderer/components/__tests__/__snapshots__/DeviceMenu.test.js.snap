// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceMenu renders a button with the proper device name 1`] = `
<span
  class="MuiButton-label-2"
>
  Keyboard
  : 
  Keyboardio Model01
</span>
`;
